### ISSUE TITLE
API Compatibility Issues

### ISSUE THREAD

**Issue by lilicoding** [tstein/AndroSS#14]
Dear developers,

We have recently developed a state-of-the-art static analysis tool for uncovering API compatibility issues in Android apps. Applying this tool to open source apps on F-droid, we have exposed a few instances of compatibility issues and submitting them to development teams for a fix. 

For your app, we have found that this project has accessed the following APIs which are available only on an API level higher than the declared minSdkVersion and which are accessed without proper protection. In other words, if those APIs get called at runtime, it will trigger a NoSuchMethodError and thus result in a crash of the running application.

<android.util.Base64: byte[] decode(java.lang.String,int)>:[8,25]
<android.util.Base64: byte[] decode(byte[],int)>:[8,25]
<android.util.Base64: byte[] encode(byte[],int,int,int)>:[8,25]
<android.util.Base64: byte[] encode(byte[],int)>:[8,25]
<android.util.Base64: byte[] decode(byte[],int,int,int)>:[8,25]

Note that, because of the nature of the static analysis, we cannot confirm whether the flagged APIs would actually be called at runtime (e.g., unreachable code). However, we still believe that those APIs, which may cause compatibility issues, should not be accessed or at least be accessed with proper protections. 

In addition to the aforementioned APIs (i.e., backward-compatibility), which could cause app crashes if accessed, we have also identified that this project has also accessed some APIs that have been removed from the latest public SDK, making the app possibly suffer from forward-compatibility issues.

NONE

We would be very much appreciated if you can acknowledge to us that those reported APIs are indeed problematic for the project’s long-term stability. please let us know if you need any more information relating to this issue report.

**Comment by tstein** [comment]
This project hasn't been updated since November 6, 2013, hasn't done anything useful since Gingerbread (2.3.7), and doesn't show up in the Play Store to Android devices running higher versions than that. You're not even right - the uses of the APIs you mention were removed in June of 2013.

You should not be opening unsolicited, automated tickets against dozens of projects like this. If you insist on doing so (and if github permits it, which I suspect it doesn't), please be sure that your reports are correct and you are opening them against maintained projects.

**Comment by lilicoding** [comment]
Dear Ted,

Thank you very much for your feedback, which is very useful for us to improve the quality of our research tool, and also the lessons we have learned.

BTW, the reason why our research tool still reports the aforementioned API issues, is as follows:
Since it is not easy for us to compile this github project, we have launched our experiment on the latest apk version of this app that is available on F-Droid.
In the version we have analyzed, there is no <uses-sdk> attribute defined (which now has defined, otheresie, to the best of our knowledge, it will likely introduce compatibility issues) and the aforementioned APIs are indeed accessed (fortunately they have been removed by now).
